configure.ac

```diff
@@ -1504,13 +1504,54 @@ case $host_os in
         ;;
 esac
 
+# MOLD section
+CUSTOM_LINKER=none
+AC_ARG_ENABLE([mold],
+               AC_HELP_STRING([--enable-mold],
```

Comment:
I thought we'd use --linker=<linked> or --linker=<linked path> so it'll be generic?

Reply:
Ah I think you missed my message on g chat (spaces). I explained why it is like this. Please have a look at it. Autoconf does not support custom args like `arg=val` And I tested all the linkers out there. `lld` and `mold` are only worth adding support for. 

Regards